### PR TITLE
Setup E2E testing with Cypress

### PR DESCRIPTION
**Pull request by dhmacs**
This PR introduces E2E testing with Cypress.

- Cypress setup
- CI setup with Github actions
- Adds tests for the newsletter subscription form

## QA

- Verify that email subscription form is sufficiently tested. In particular the user should see an error when clicking subscribe and the email is missing or invalid, and he should see if the request has been successful or a server error occurred.
- Verify also that all tests are correctly processed by CI and have passed (see github action logs).

### CODE REVIEW

File: cypress/integration/collection/newsletter_subscribe.spec.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      user.findByPlaceholderText(/enter your email/i).type('test@example');
+      user.findByRole('button', { name: /subscribe$/i }).click();
+      user
+         .findByRole('button', { name: /subscribe$/i })
+         .parent('form')
+         .contains(/insert a valid email/i);
+   });
+
+   it('shows confirmation when email is subscribed', () => {
+      cy.intercept('/api/2.0/cart/newsletter/subscribe', {
```
Comment (mlahargou) [inline cypress/integration/collection/newsletter_subscribe.spec.ts]: This is cool!

File: cypress/integration/collection/newsletter_subscribe.spec.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+   it('requires an email', () => {
+      user.findByRole('button', { name: /subscribe$/i }).click();
+      user
+         .findByRole('button', { name: /subscribe$/i })
+         .parent('form')
+         .contains(/insert a valid email/i);
+   });
+
+   it('prevents invalid email', () => {
+      user.findByPlaceholderText(/enter your email/i).type('test@example');
```
Comment (mlahargou) [inline cypress/integration/collection/newsletter_subscribe.spec.ts]: Where is findByPlaceholderText defined? I don't see it in the cypress docs.

**Comment by mlahargou** [comment]
CR 📱

**Comment by sterlinghirsh** [comment]
CR :zap:

File: cypress/integration/collection/newsletter_subscribe.spec.ts (hunk 2 above)
Comment (dhmacs) [reply cypress/integration/collection/newsletter_subscribe.spec.ts]: @mlahargou I'm using [Testing Library](https://testing-library.com/docs/cypress-testing-library/intro) which adds a few useful utils for testing, and helps create tests that reads more like what the user is doing.

[Here](https://github.com/iFixit/react-commerce/pull/23/files#diff-807c668e3005f98d28ed29b2f43426bbc52ccaba3e5e113f8a4451244b4cf3f8R26) you can see where this is setup.

**Comment by batbattur** [comment]
**Steps used to test:** 
1. cloned the repo locally
2. checked out this branch
3. `npm install && npm run dev` 
4.  visited `http://localhost:3000/collections/mac-desktop` and checked the subscription form with valid/invalid emails. ✅ 
5. `npx cypress open`
5. Ran the cypress test and made sure it tests the subscription form sufficiently ✅ 

Screen Shot 2021-06-16 at 9 08 00 AM

6. Verified that the CI run for cypress in [github action logs ✅ ](https://github.com/iFixit/react-commerce/pull/23/checks?check_run_id=2831090830)
Screen Shot 2021-06-16 at 9 20 18 AM


QA 👍

**Pull request merged**